// plugin-welcome/src/api/models/ControllersScholarship.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface ControllersScholarship
 */
export interface ControllersScholarship {
    /**
     * 
     * @type {number}
     * @memberof ControllersScholarship
     */
    educationLevelID?: number;
    /**
     * 
     * @type {string}
     * @memberof ControllersScholarship
     */
    organization?: string;
    /**
     * 
     * @type {string}
     * @memberof ControllersScholarship
     */
    scholarshipName?: string;
    /**
     * 
     * @type {number}
     * @memberof ControllersScholarship
     */
    scholarshipTypeID?: number;
    /**
     * 
     * @type {number}
     * @memberof ControllersScholarship
     */
    semesterID?: number;
    /**
     * 
     * @type {number}
     * @memberof ControllersScholarship
     */
    studyPlanID?: number;
    /**
     * 
     * @type {number}
     * @memberof ControllersScholarship
     */
    userID?: number;
}

export function ControllersScholarshipFromJSON(json: any): ControllersScholarship {
    return ControllersScholarshipFromJSONTyped(json, false);
}

export function ControllersScholarshipFromJSONTyped(json: any, ignoreDiscriminator: boolean): ControllersScholarship {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'educationLevelID': !exists(json, 'educationLevelID') ? undefined : json['educationLevelID'],
        'organization': !exists(json, 'organization') ? undefined : json['organization'],
        'scholarshipName': !exists(json, 'scholarshipName') ? undefined : json['scholarshipName'],
        'scholarshipTypeID': !exists(json, 'scholarshipTypeID') ? undefined : json['scholarshipTypeID'],
        'semesterID': !exists(json, 'semesterID') ? undefined : json['semesterID'],
        'studyPlanID': !exists(json, 'studyPlanID') ? undefined : json['studyPlanID'],
        'userID': !exists(json, 'userID') ? undefined : json['userID'],
    };
}

export function ControllersScholarshipToJSON(value?: ControllersScholarship | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'educationLevelID': value.educationLevelID,
        'organization': value.organization,
        'scholarshipName': value.scholarshipName,
        'scholarshipTypeID': value.scholarshipTypeID,
        'semesterID': value.semesterID,
        'studyPlanID': value.studyPlanID,
        'userID': value.userID,
    };
}
