// plugin-welcome/src/api/models/EntSemesterEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntScholarship,
    EntScholarshipFromJSON,
    EntScholarshipFromJSONTyped,
    EntScholarshipToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntSemesterEdges
 */
export interface EntSemesterEdges {
    /**
     * SemesterID holds the value of the SemesterID edge.
     * @type {Array<EntScholarship>}
     * @memberof EntSemesterEdges
     */
    semesterID?: Array<EntScholarship>;
}

export function EntSemesterEdgesFromJSON(json: any): EntSemesterEdges {
    return EntSemesterEdgesFromJSONTyped(json, false);
}

export function EntSemesterEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntSemesterEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'semesterID': !exists(json, 'semesterID') ? undefined : ((json['semesterID'] as Array<any>).map(EntScholarshipFromJSON)),
    };
}

export function EntSemesterEdgesToJSON(value?: EntSemesterEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'semesterID': value.semesterID === undefined ? undefined : ((value.semesterID as Array<any>).map(EntScholarshipToJSON)),
    };
}
