// plugin-welcome/src/api/models/EntStudyplanEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntScholarship,
    EntScholarshipFromJSON,
    EntScholarshipFromJSONTyped,
    EntScholarshipToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntStudyplanEdges
 */
export interface EntStudyplanEdges {
    /**
     * StudyplanID holds the value of the StudyplanID edge.
     * @type {Array<EntScholarship>}
     * @memberof EntStudyplanEdges
     */
    studyplanID?: Array<EntScholarship>;
}

export function EntStudyplanEdgesFromJSON(json: any): EntStudyplanEdges {
    return EntStudyplanEdgesFromJSONTyped(json, false);
}

export function EntStudyplanEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntStudyplanEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'studyplanID': !exists(json, 'studyplanID') ? undefined : ((json['studyplanID'] as Array<any>).map(EntScholarshipFromJSON)),
    };
}

export function EntStudyplanEdgesToJSON(value?: EntStudyplanEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'studyplanID': value.studyplanID === undefined ? undefined : ((value.studyplanID as Array<any>).map(EntScholarshipToJSON)),
    };
}
