// plugin-welcome/src/api/models/EntStudyplan.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntStudyplanEdges,
    EntStudyplanEdgesFromJSON,
    EntStudyplanEdgesFromJSONTyped,
    EntStudyplanEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntStudyplan
 */
export interface EntStudyplan {
    /**
     * Studyplanname holds the value of the "Studyplanname" field.
     * @type {string}
     * @memberof EntStudyplan
     */
    studyplanname?: string;
    /**
     * 
     * @type {EntStudyplanEdges}
     * @memberof EntStudyplan
     */
    edges?: EntStudyplanEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntStudyplan
     */
    id?: number;
}

export function EntStudyplanFromJSON(json: any): EntStudyplan {
    return EntStudyplanFromJSONTyped(json, false);
}

export function EntStudyplanFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntStudyplan {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'studyplanname': !exists(json, 'Studyplanname') ? undefined : json['Studyplanname'],
        'edges': !exists(json, 'edges') ? undefined : EntStudyplanEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
    };
}

export function EntStudyplanToJSON(value?: EntStudyplan | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'Studyplanname': value.studyplanname,
        'edges': EntStudyplanEdgesToJSON(value.edges),
        'id': value.id,
    };
}
